bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/protocol/EmbyClientSocket.java

```diff
@@ -0,0 +1,209 @@
+ * @author Zachary Christiansen - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class EmbyClientSocket {
+
+    private static final int REQUEST_TIMEOUT_MS = 10000;
+    private final Logger logger = LoggerFactory.getLogger(EmbyClientSocket.class);
+    private final ScheduledExecutorService scheduler;
+    private final Gson mapper = new Gson();
+    private final EmbyClientSocketEventListener eventHandler;
+    private final WebSocketClient client;
+    private final ClientUpgradeRequest request = new ClientUpgradeRequest();
+    private final EmbyWebSocketListener socket = new EmbyWebSocketListener();
+
+    private @Nullable URI uri;
+    private @Nullable Session session;
+    private int bufferSize;
+
+    private volatile boolean shouldReconnect = true;
+    private int reconnectAttempts = 0;
+
+    public EmbyClientSocket(EmbyClientSocketEventListener handler, @Nullable URI setUri,
+            ScheduledExecutorService setScheduler, int setBufferSize, WebSocketClient sharedWebSocketClient) {
+        this.eventHandler = handler;
+        this.uri = setUri;
+        this.scheduler = setScheduler;
+        this.bufferSize = setBufferSize;
+        this.client = sharedWebSocketClient;
+    }
+
+    public synchronized void open() throws Exception {
+        if (isConnected()) {
+            logger.debug("already open");
+            return;
+        }
+        if (!client.isStarted()) {
+            client.setMaxTextMessageBufferSize(bufferSize);
+        }
+        Future<Session> future = requireNonNull(client.connect(socket, uri, request),
+                "WebSocketClient.connect returned null Future<Session>");
+        Session wsSession = requireNonNull(future.get(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS),
+                "Future<Session>.get returned null Session");
+        requireNonNull(wsSession, "WebSocketClient.connect returned null Session");
+        this.session = wsSession;
+        logger.info("Connected to Emby");
+        // Reset retry counters
+        reconnectAttempts = 0;
+        // Fire the connection‐opened event
+        scheduler.submit(() -> eventHandler.onConnectionOpened());
+    }
+
+    public void close() {
+        shouldReconnect = false;
+        reconnectAttempts = 0;
+        Session s = session;
+        if (s != null) {
+            try {
+                s.close();
+            } catch (Exception e) {
+                logger.debug("Exception during closing the websocket: {}", e.getMessage(), e);
+            }
+            session = null;
+        }
+    }
+
+    public boolean isConnected() {
+        Session s = session;
+        return s != null && s.isOpen();
+    }
+
+    public synchronized void attemptReconnect() {
+        if (!shouldReconnect) {
+            return;
+        }
+        reconnectAttempts++;
+        long delay = Math.min(60_000, (1 << Math.min(reconnectAttempts, 6)) * 1000L);
+        logger.info("Scheduling reconnect #{} in {}ms", reconnectAttempts, delay);
```

Comment:
info level should be used as less as possible. only iof really import to the user
```suggestion
        logger.debug("Scheduling reconnect #{} in {}ms", reconnectAttempts, delay);
```

Reply:
changed